servers/rendering/renderer_rd/shaders/scene_forward_clustered.glsl

```diff
@@ -969,6 +969,13 @@ void fragment_shader(in SceneData scene_data) {
 #endif //not render depth
 	/////////////////////// LIGHTING //////////////////////////////
 
+// FIXME: Run this block only if the shader writes into the VERTEX variable
+#ifdef USE_MULTIVIEW
+	view = -normalize(vertex_interp - scene_data.eye_offset[ViewIndex].xyz);
+#else
+	view = -normalize(vertex_interp);
+#endif
+
```

Comment:
These lines should not have any effect. This code is already being run on line 652, immediately inside `void fragment_shader`.

`vertex_interp` is a `in vec3` parameter, so it is not related to the local `vertex` variable that is being written by your shader.

If a shader change is needed, you will need to update scene_forward_mobile.glsl as well as possibly the shaders in `drivers/glsl3`. But I'm pretty sure that if you can remove this code, nothing will change.

Reply:
It supposed to be 
```GLSL
#ifdef USE_MULTIVIEW
	view = -normalize(vertex - scene_data.eye_offset[ViewIndex].xyz);
#else
	view = -normalize(vertex);
#endif
````
(and the other variant) which I fixed in the branch.

Is the shader code under drivers/glsl3 still in use? If yes, then in which case?

Yes, the mobile GLSL code will be updated as well once we settle with a final solution.